pkg/nodejs/pnpm/parse.go

```diff
@@ -95,15 +116,61 @@ func (p *Parser) parse(lockFile *LockFile) ([]types.Library, []types.Dependency)
 	return libs, deps
 }
 
-func isIndirectLib(name string, directDeps map[string]string) bool {
+func isIndirectLib(name string, directDeps map[string]interface{}) bool {
 	_, ok := directDeps[name]
 	return !ok
 }
 
-func getPackageNameAndVersion(pkg string) (string, string) {
-	idx := strings.LastIndex(pkg, "/")
-	name := pkg[1:idx]
-	version := pkg[idx+1:]
-
+func getPackageNameAndVersion(pkg, name, version string, lockFileVersion float64) (string, string) {
+	// local archives have `name` and `version` fields
+	// https://github.com/pnpm/spec/blob/ad27a225f81d9215becadfa540ef05fa4ad6dd60/lockfile/5.2.md#packagesdependencypathname
+	if strings.HasPrefix(pkg, "file:") {
```

Comment:
I thought archives started with `@` like `@registry.npmjs.org/is-positive/-/is-positive-1.0.0.tgz`
https://github.com/pnpm/pnpm/blob/ce61f8d3c29eee46cee38d56ced45aea8a439a53/lockfile/lockfile-file/test/lockfileV6Converters.test.ts#LL164C7-L164C63

Where is the `file:` prefix defined?

Reply:
there are 3 cases:
- local archive
- git
- remove archive

`pnpm` parses that [here](https://github.com/pnpm/pnpm/blob/04a279881a47b0d7b5a32a5bc774d0ff071d1e2f/fetching/pick-fetcher/src/index.ts#L7C1-L14)

There is also install from github. This is also archive,  but there is no information about it in the package name (only `resolution` field says that it is archive).:
```
  github.com/debug-js/debug/d1616622e4d404863c5a98443f755b4006e971dc:
    resolution: {tarball: https://codeload.github.com/debug-js/debug/tar.gz/d1616622e4d404863c5a98443f755b4006e971dc}
    name: debug
    version: 4.3.4
    engines: {node: '>=6.0'}
    peerDependencies:
      supports-color: '*'
    peerDependenciesMeta:
      supports-color:
        optional: true
    dependencies:
      ms: 2.1.2
    dev: false
```

all these cases (archives and github) have `name` and `version` fields.
Looks like it's wrong to parse  archive name, because archive can use any name (look this -https://github.com/aquasecurity/go-dep-parser/pull/220/files#diff-fc1d9989ef2210b568c552d504211fa3ff47735ea7b17865f3942921d516d89cR149 you can use any name, but `name` and `version` field will be correct.):
```
  file:foo/bar/lodashhhh-any-version.tgz:
    resolution: {integrity: sha512-x06dhjWXFr2GpUpY0LyzYOAUIppa/T/PpY4O7qu9kIAJESqDjOqTur552ggOOfaZNdAervnxmXpFJIRGULsl+w==, tarball: file:foo/bar/lodashhhh-any-version.tgz}
    name: lodash
    version: 4.17.10
    dev: false

```
I think we can use next logic:
1) check `name` and `version` fields (we will parse archives and packages installed from github)
2) if these fields are empty - use `getPackageNameAndVersion` function.

@knqyf263 wdyt?

Reply:
we can find tarballs from `Resolution.Tarball` field. Take a look on this commit - https://github.com/aquasecurity/go-dep-parser/pull/220/commits/584f3c9a53e5ec77c04c9c54ad430482abc80361

Reply:
Looks good.